compiler/src/yul/runtime/functions.rs

```diff
@@ -62,39 +64,104 @@ pub fn free() -> yul::Statement {
 }
 
 /// Copy calldata to memory a newly allocated segment of memory.
-pub fn ccopy() -> yul::Statement {
+pub fn calldata_to_mem() -> yul::Statement {
     function_definition! {
-        function ccopy(cptr, size) -> mptr {
+        function calldata_to_mem(cptr, size) -> mptr {
             (mptr := alloc(size))
             (calldatacopy(mptr, cptr, size))
         }
     }
 }
 
 /// Copy memory to a given segment of storage.
-pub fn mcopy() -> yul::Statement {
+pub fn mem_to_sto() -> yul::Statement {
     function_definition! {
-        function mcopy(mptr, sptr, size) {
-            (for {(let i := 0)} (lt(i, size)) {(i := add(i, 1))}
+        function mem_to_sto(mptr, sptr, size) {
+            (let offset := 0)
+            (for { } (lt((add(offset, 32)), size)) { }
             {
-                (let _mptr := add(mptr, i))
-                (let _sptr := add(sptr, i))
-                (sstoren(_sptr, (mloadn(_mptr, 1)), 1))
+                (let _mptr := add(mptr, offset))
+                (let _sptr := add(sptr, offset))
+                (sstore(_sptr, (mload(_mptr))))
+                (offset := add(offset, 32))
+            })
+
+            (let rem := sub(size, offset))
+            (if (gt(rem, 0)) {
+                (let _mptr := add(mptr, offset))
+                (let _sptr := add(sptr, offset))
+                (sstoren(_sptr, (mloadn(_mptr, rem)), rem))
```

Comment:
Suggestion - All these functions have a similar definition the only difference I can see is the load operation & store operation. Can't we use macros to reduce the code dedup?

Reply:
just tried my hand at it, but realized there's the difficulty of dealing with the different signatures on `sstore` and `mstore`